data-prepper-core/src/test/java/com/amazon/dataprepper/parser/config/MetricsConfigTest.java

```diff
@@ -172,6 +179,41 @@ public void testGivenConfigWithEMFLoggingMeterRegistryThenMeterRegistryCreated()
         assertThat(commonTag.getValue(), equalTo(DataPrepper.getServiceNameForMetrics()));
     }
 
+    @ParameterizedTest
+    @MethodSource("provideMetricRegistryTypesAndCreators")
+    public void testGivenConfigWithMetricTagsThenMeterRegistryConfigured(final MetricRegistryType metricRegistryType,
+                                                                         final Function<DataPrepperConfiguration, MeterRegistry> creator) {
+        final String testKey = "testKey";
+        final String testValue = "testValue";
+        final String testServiceName = "testServiceName";
+        final DataPrepperConfiguration dataPrepperConfiguration = mock(DataPrepperConfiguration.class);
+        when(dataPrepperConfiguration.getMetricRegistryTypes()).thenReturn(Collections.singletonList(metricRegistryType));
+        when(dataPrepperConfiguration.getMetricTags()).thenReturn(Map.of(testKey, testValue));
+
+        MeterRegistry meterRegistry = creator.apply(dataPrepperConfiguration);
+        Counter counter = meterRegistry.counter("counter");
+        List<Tag> commonTags = counter.getId().getConventionTags(meterRegistry.config().namingConvention());
+
+        assertThat(commonTags.equals(
```

Comment:
Why not?

```
assertThat(commonTags, equalTo(Arrays.asList(...))
```

This approach will give more information if the objects are not equal, making it easier to fix problems.